iceoryx_dust/container/include/iox/detail/fixed_position_container.inl

```diff
@@ -94,73 +82,79 @@ FixedPositionContainer<T, CAPACITY>::operator=(FixedPositionContainer&& rhs) noe
 {
     if (this != &rhs)
     {
-        copy_and_move_impl(std::move(rhs));
-
-        // clear rhs
-        rhs.clear();
+        copy_and_move_impl<detail::SpecialCreationOperations::MoveAssignment>(std::move(rhs));
     }
     return *this;
 }
 
 template <typename T, uint64_t CAPACITY>
-template <typename RhsType>
+template <detail::SpecialCreationOperations Opt, typename RhsType>
 inline void FixedPositionContainer<T, CAPACITY>::copy_and_move_impl(RhsType&& rhs) noexcept
 {
-    // we already make sure rhs is always passed by std::move() for move case, therefore
-    // the result of "decltype(rhs)" is same as "decltype(std::forward<RhsType>(rhs))"
-    static_assert(std::is_rvalue_reference<decltype(rhs)>::value
-                      || (std::is_lvalue_reference<decltype(rhs)>::value
-                          && std::is_const<std::remove_reference_t<decltype(rhs)>>::value),
-                  "RhsType must be const lvalue reference or rvalue reference");
+    static_assert(
+        std::is_rvalue_reference<decltype(rhs)>::value
+            || (std::is_lvalue_reference_v<decltype(rhs)> && std::is_const_v<std::remove_reference_t<decltype(rhs)>>),
+        "RhsType must be const lvalue reference or rvalue reference");
```

Comment:
I wonder whether there is a nice way to move this to `SpecialCreationHelper`?

Reply:
I think this can be remove directly, since we also check this properties in `ctor_create` and `assignment_create`. What do you think :)

Reply:
Yes, remove it. No need to do the check multiple times.